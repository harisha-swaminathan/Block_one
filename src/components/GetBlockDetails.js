import React from 'react'
import * as Eos from 'eosjs'
import EachBlockItem from './EachBlockItem'


export default class GetBlockDetails extends React.Component{
    constructor(){
        super()
        this.state={
            block:[],
            actions:[]
        }
    }
    async fetchRecentBlocks(){
        const blocks=[]
        let block
        let actions=[]
        let action
        const rpc = new Eos.JsonRpc('http://127.0.0.1:8888', {});
        const info= await rpc.get_info()
        let id=info.head_block_num
        for(let i=0;i<10;i++){
            block= await rpc.get_block(id)

            action= await rpc.history_get_actions(block.producer)
            // block['action']=action
            actions.push(action.actions)
           
            
            blocks.push(block)
            id=block.previous
        }
        this.setState({block:blocks, actions})

    }
    onLoadClick=()=>{
        this.fetchRecentBlocks()
    }
    componentDidMount(){
        this.fetchRecentBlocks()
        }

    render(){
        return(
        <div className="container">
            <button onClick={this.onLoadClick} className="block__button">Load recent</button>
            <div className="block">
                    {this.state.block.length!==0 && 
                        <div>
        
                            {this.state.block.map((block, i)=>{
                            return <EachBlockItem key={block.id} block={block} actions={this.state.actions[i]}/>})}
                        </div>
                    }
            </div>
        </div> 
        )
    }
}