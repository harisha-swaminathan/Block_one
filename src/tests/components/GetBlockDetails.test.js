import React from 'react';
import { shallow } from 'enzyme';
import GetBlockDetails from '../../components/GetBlockDetails'
const Eos = jest.requireActual('eosjs')

test('renders the component correctly ', () => {
    const wrapper = shallow(<GetBlockDetails />);
    expect(wrapper.debug()).toMatchSnapshot();
});

test('calls onClick function on button click  ', () => {
    const onClickMock = jest.fn();
    GetBlockDetails.prototype.fetchRecentBlocks=onClickMock
    const wrapper = shallow(<GetBlockDetails />);
    wrapper.find('button').simulate('click')
    expect(onClickMock).toHaveBeenCalled()
});