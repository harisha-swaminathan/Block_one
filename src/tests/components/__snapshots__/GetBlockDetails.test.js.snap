// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly  1`] = `
"<div className=\\"container\\">
  <button onClick={[Function: onClick]} className=\\"block__button\\">
    Load recent
  </button>
  <div className=\\"block\\" />
</div>"
`;
